airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -322,10 +324,17 @@ def read(self, fields: Iterable = None) -> Iterable[dict]:
             if self._reader_format == "json" or self._reader_format == "jsonl":
                 yield from self.load_nested_json(fp)
             else:
+                fields = list(fields)
                 for df in self.load_dataframes(fp):
-                    columns = set(fields).intersection(set(df.columns)) if fields else df.columns
-                    df = df.replace(np.nan, "NaN", regex=True)
-                    yield from df[columns].to_dict(orient="records")
+                    if self._reader_format == "csv" and isinstance(df, pd.io.parsers.TextFileReader):
```

Comment:
```python
chunks = []
if self._reader_format == "csv" and isinstance(df, pd.io.parsers.TextFileReader):
    chunks = df
```

Reply:
if next suggestion applied then this one don't needed

Reply:
For now, I implemented this version. Let's wait for @vitaliizazmic's review also.

Reply:
I think, it's better to change load_dataframes method

Reply:
Ok.